test/ci/kokoro/linux/run_integ_tests.sh

```diff
@@ -2,33 +2,37 @@
 
 # This shell script is used for setting up our Kokoro Ubuntu environment
 # with necessary dependencies for running integration tests, and then
-# running tests when PRs are submitted.
+# running tests when PRs are submitted or code is pushed.
+#
+# This script is intentionally a bit verbose with what it writes to stdout.
+# Since its output is only visible in test run logs, and those logs are only
+# likely to be viewed in the event of an error, I decided it would be beneficial
+# to leave some settings like `set -x` and `cat`s and `echo`s in. This should
+# help future engineers debug what went wrong, and assert info about the test
+# environment at the cost of a small preamble in the logs.
 
-# For now, continuous.sh and presubmit.sh are both symlinks to this file.
-# Kokoro looks for files with those names, but our continuous and presubmit jobs
-# should be identical on Linux.
-
-# -e : Fail on any error
 # -x : Display commands being run
 # -u : Disallow unset variables
 # Doc: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html#The-Set-Builtin
-set -exu
-
-GITHUB_REPO="https://github.com/GoogleCloudPlatform/gsutil"
-GSUTIL_KEY="/src/keystore/gsutil_kokoro_service_key.json"
-GSUTIL_SRC_PATH="/src/gsutil"
-GSUTIL_ENTRYPOINT="$GSUTIL_SRC_PATH/gsutil.py"
-PYTHON_PATH="/usr/local/bin/python"
-CONFIG_JSON="/src/.boto_json"
-CONFIG_XML="/src/.boto_xml"
+set -xu
 
-# Processes to use based on default Ubuntu Kokoro specs here:
-# go/gcp-ubuntu-vm-configuration-v32i
-PROCS="4"
 
-# PYMAJOR and PYMINOR environment variables are set for each Kokoro job in:
+# PYMAJOR, PYMINOR, and BOTO_CONFIG environment variables are set per job in:
 # go/kokoro-gsutil-configs
 PYVERSION="$PYMAJOR.$PYMINOR"
+API=${BOTO_CONFIG##*_}
```

Comment:
What on earth is this `##*_` stuff? Lol

Reply:
I'll add a comment! It uses [parameter expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) matches the string after the rightmost instance of `_` in the BOTO_CONFIG environment variable.

So `/tmpfs/src/.boto_json` yields `json`. Alternatively, we can change to variable to just be `xml` or `json`, then append a path to it in the test runner script here, albeit we'll also need to change all the configs.

Typing this out, the latter sounds more proper. Let me know if you have any preference. 👍

Reply:
Updated the environment variables in cl/239702660 and the most recent commit here.